storage/bucket_test.go

```diff
@@ -663,3 +663,31 @@ func TestNewBucket(t *testing.T) {
 		t.Errorf("got=-, want=+:\n%s", diff)
 	}
 }
+
+func TestObjectsWithProjection(t *testing.T) {
```

Comment:
Instead of this, can we do an integration test like what we have for SetAttrSelection? https://github.com/googleapis/google-cloud-go/blob/master/storage/integration_test.go#L1174